internal/service/generate/errordefinition/generator/generate.go

```diff
@@ -0,0 +1,116 @@
+package generator
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/dave/jennifer/jen"
+
+	"github.com/pace/bricks/http/jsonapi/runtime"
+)
+
+const (
+	pkgBricksErrors = "github.com/pace/bricks/maintenance/errors"
+)
+
+// Generator for go error types for the given json api error definition.
+type Generator struct {
+	goSource    *jen.File
+	serviceName string
+}
+
+func loadDefinitionDataFromURI(url *url.URL) ([]byte, error) {
+	resp, err := http.Get(url.String())
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close() // nolint: errcheck
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+// BuildSource generates the go code in the specified path with specified package name
+// based on the passed schema source (url or file path)
+func (g *Generator) BuildSource(source, packagePath, packageName string) (string, error) {
+
+	var data []byte
+	if strings.HasPrefix(source, "http://") || strings.HasPrefix(source, "https://") {
+		loc, err := url.Parse(source)
+		if err != nil {
+			return "", err
+		}
+
+		data, err = loadDefinitionDataFromURI(loc)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		// read definition file from disk
+		var err error
+		data, err = os.ReadFile(source) // nolint: gosec
+		if err != nil {
+			return "", err
+		}
+
+	}
+
+	// parse definition
+	var errors runtime.Errors
+	err := json.Unmarshal(data, &errors)
+	if err != nil {
+		return "", err
+	}
+
+	return g.BuildDefinitions(errors, packagePath, packageName)
+}
+
+func (g *Generator) BuildDefinitions(errors runtime.Errors, packagePath, packageName string) (string, error) {
+
+	g.goSource = jen.NewFilePathName(packagePath, packageName)
+	g.goSource.PackageComment("// Code generated by github.com/pace/bricks DO NOT EDIT.")
+
+	g.serviceName = packageName
+
+	for _, jsonError := range errors {
+
+		// skip example if given
+		if jsonError.Code == "EXAMPLE" {
+			continue
+		}
+
+		httpStatusCode, err := strconv.Atoi(jsonError.Status)
+		if err != nil {
+			return "", fmt.Errorf("%w: unable to parse given http status code")
+		}
+
+		g.goSource.Func().Id(fmt.Sprintf("%s", jsonError.Code)).Params(
```

Comment:
Maybe the function name should be prefixed with `ERR_`, or `ErrAndCodeAsCamelCase()`??

It makes it clearer what that function is if it's prefixed like that.

Reply:
Add the proposed `ERR_` prefix.